ui/css-pages-widgets.md

```diff
@@ -0,0 +1,87 @@
+---
+
+# Styling pages & widgets using CSS
+
+## Predefined CSS Classes
+
+As seen before, you can use CSS classes in the `class` property (as an array) or set CSS properties in the `style` property (as a dictionary) on your components.
+
+You cannot define new CSS classes, but you can use classes from Framework7, for instance:
+
+- [Typography](https://v5.framework7.io/docs/typography.html)
+- [Color Themes](https://v5.framework7.io/docs/color-themes.html#apply-color-themes)
+- [Hairlines](https://v5.framework7.io/docs/hairlines.html)
+- [Elevation](https://v5.framework7.io/docs/elevation.html)
+
+## CSS Variables
+
+Another interesting technique is to override the many [CSS Variables](https://v5.framework7.io/docs/css-variables.html) defined by Framework7 to suit your particular component's needs.
+The override will be applied to all descendants in the HTML DOM tree, in case of the Framework7 variables the underlying components which use them will use the new value.
+It is recommended to use Framework7 CSS variables in your components too, when appropriate, so that way you'll be sure the properties change according to the current theme and dark mode setting.
+
+To redefine a CSS variable for the component and its descendants, use this syntax:
+
+```yaml
+style:
+  --f7-button-border-color: rgb(255, 0, 0)
+```
+
+To reuse a CSS variable, use:
+
+```yaml
+  border-color: var(--f7-button-border-color)
+```
+
+You can even define your own CSS variables and use them in your components:
+
+```yaml
+config:
+  style:
+    --my-color: =props.color
+slots:
+  ...
+     ...
+        ...
+           config:
+             style:
+               background-color: var(--my-color)
+```
+
+## Applying CSS Properties Directly
+
+Applying CSS properties like `border-color` directly on components is sometimes enough; but contrary to CSS variables like `--f7-button-border-color` which will be inherited to descendants in the tree, either by your own components or by f7 components (or their OH derivatives) that make use of these variables, they will only work on the element where you put the style configuration.
+
+There are hundreds of [CSS properties](https://www.w3schools.com/cssref/) you can use to design your widgets.
+Use the resources at [W3Schools](https://www.w3schools.com/css/default.asp) or the [CSS-Tricks Properties Almanac](https://css-tricks.com/almanac/properties/) to learn more about CSS properties and techniques - these resources will provide code in HTML and CSS class definitions but you can most of the time adapt them for usage in components' YAML definitions.
+
+While openHAB's layout components (`oh-block`, `oh-grid-row`, `oh-grid-col`) can help you with the placement of your widgets, to layout sub-components _within_ a widget, you shouldn't use them because they include design mode controls that you don't need.
+While you can use their `f7-block`, `f7-row` and `f7-col` equivalents instead, in many cases this is still "overkill":
```

Comment:
While twice doesn't sound that nice. Maybe "Though" here?

Reply:
Good suggestion, thx!

Reply:
Commited :+1: